File: java/src/test/java/io/binac/leetcode/CombinationSumIITest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;

import static org.junit.jupiter.api.Assertions.*;

class CombinationSumIITest {
    private static final CombinationSumII.Solution1 SOLUTION1 = new CombinationSumII.Solution1();

    private final int[] CANDIDATES1 = {10, 1, 2, 7, 6, 1, 5};
    private final int TARGET1 = 8;
    private final List<List<Integer>> EXPECTED1 = Arrays.asList(
            Arrays.asList(1, 1, 6),
            Arrays.asList(1, 2, 5),
            Arrays.asList(1, 7),
            Arrays.asList(2, 6)
    );

    private final int[] CANDIDATES2 = {2, 5, 2, 1, 2};
    private final int TARGET2 = 5;
    private final List<List<Integer>> EXPECTED2 = Arrays.asList(
            Arrays.asList(1, 2, 2),
            Collections.singletonList(5)
    );

    @Test
    void test1() {
        assertEquals(EXPECTED1, SOLUTION1.combinationSum2(CANDIDATES1, TARGET1));
        assertEquals(EXPECTED2, SOLUTION1.combinationSum2(CANDIDATES2, TARGET2));
    }
}

File: cpp/unittests/CombinationSumIITest.cpp
#include "CombinationSumII.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(CombinationSumII, Solution40_1) {
  std::vector<int> Candidates1{10, 1, 2, 7, 6, 1, 5};
  const int Target1 = 8;
  const std::vector<std::vector<int>> Expected1{
      {1, 1, 6},
      {1, 2, 5},
      {1, 7},
      {2, 6}
  };
  EXPECT_EQ(Expected1, Solution40_1().combinationSum2(Candidates1, Target1));

  std::vector<int> Candidates2{2, 5, 2, 1, 2};
  const int Target2 = 5;
  const std::vector<std::vector<int>> Expected2{
      {1, 2, 2},
      {5},
  };
  EXPECT_EQ(Expected2, Solution40_1().combinationSum2(Candidates2, Target2));
}